app/RequestLoggingFilter.scala

```diff
@@ -0,0 +1,67 @@
+import akka.stream.Materializer
+import net.logstash.logback.marker.Markers.appendEntries
+import play.api.mvc.{Filter, RequestHeader, Result}
+import play.api.MarkerContext
+import utils.Logging
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.jdk.CollectionConverters._
+import scala.util.{Failure, Success}
+
+class RequestLoggingFilter(override val mat: Materializer)(implicit ec: ExecutionContext) extends Filter with Logging {
+  override def apply(next: (RequestHeader) => Future[Result])(rh: RequestHeader): Future[Result] = {
+    val start = System.currentTimeMillis()
+    val result = next(rh)
+
+    result onComplete {
+      case Success(response) =>
+        val duration = System.currentTimeMillis() - start
+        logSuccess(rh, response, duration)
+
+      case Failure(err) =>
+        val duration = System.currentTimeMillis() - start
+        logFailure(rh, err, duration)
+    }
+
+    result
+  }
+
+  private def logSuccess(request: RequestHeader, response: Result, duration: Long): Unit = {
+    val originIp = request.headers.get("X-Forwarded-For").getOrElse(request.remoteAddress)
+    val referer = request.headers.get("Referer").getOrElse("")
+    val userAgent = request.headers.get("User-Agent").getOrElse("")
```

Comment:
Can be used to distinguish human vs. machine users. For example, a user-agent of "node-fetch" is a machine.